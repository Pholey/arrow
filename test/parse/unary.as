// Identity
+43;
++1;

// Negate
-1;
--26;

// BitNot
~x;
~~y;

// Not
not x;
not y < x;

// ~
+-~--y;
