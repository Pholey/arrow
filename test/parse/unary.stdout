{
  "tag": "Module",
  "span": "test/parse/unary.as:1:1-1",
  "statements": [
    {
      "tag": "Identity",
      "span": "test/parse/unary.as:2:1-4",
      "operand": {
        "tag": "Integer",
        "span": "test/parse/unary.as:2:2-4",
        "value": "43"
      }
    },
    {
      "tag": "Identity",
      "span": "test/parse/unary.as:3:1-4",
      "operand": {
        "tag": "Identity",
        "span": "test/parse/unary.as:3:2-4",
        "operand": {
          "tag": "Integer",
          "span": "test/parse/unary.as:3:3",
          "value": "1"
        }
      }
    },
    {
      "tag": "Negate",
      "span": "test/parse/unary.as:6:1-3",
      "operand": {
        "tag": "Integer",
        "span": "test/parse/unary.as:6:2",
        "value": "1"
      }
    },
    {
      "tag": "Negate",
      "span": "test/parse/unary.as:7:1-5",
      "operand": {
        "tag": "Negate",
        "span": "test/parse/unary.as:7:2-5",
        "operand": {
          "tag": "Integer",
          "span": "test/parse/unary.as:7:3-5",
          "value": "26"
        }
      }
    },
    {
      "tag": "BitNot",
      "span": "test/parse/unary.as:10:1-3",
      "operand": {
        "tag": "Identifier",
        "span": "test/parse/unary.as:10:2",
        "text": "x"
      }
    },
    {
      "tag": "BitNot",
      "span": "test/parse/unary.as:11:1-4",
      "operand": {
        "tag": "BitNot",
        "span": "test/parse/unary.as:11:2-4",
        "operand": {
          "tag": "Identifier",
          "span": "test/parse/unary.as:11:3",
          "text": "y"
        }
      }
    },
    {
      "tag": "Not",
      "span": "test/parse/unary.as:14:1-6",
      "operand": {
        "tag": "Identifier",
        "span": "test/parse/unary.as:14:5",
        "text": "x"
      }
    },
    {
      "tag": "Not",
      "span": "test/parse/unary.as:15:1-6",
      "operand": {
        "tag": "LessThan",
        "span": "test/parse/unary.as:15:7-6",
        "lhs": {
          "tag": "Identifier",
          "span": "test/parse/unary.as:15:5",
          "text": "y"
        },
        "rhs": {
          "tag": "Identifier",
          "span": "test/parse/unary.as:15:9",
          "text": "x"
        }
      }
    },
    {
      "tag": "Identity",
      "span": "test/parse/unary.as:18:1-7",
      "operand": {
        "tag": "Negate",
        "span": "test/parse/unary.as:18:2-7",
        "operand": {
          "tag": "BitNot",
          "span": "test/parse/unary.as:18:3-7",
          "operand": {
            "tag": "Negate",
            "span": "test/parse/unary.as:18:4-7",
            "operand": {
              "tag": "Negate",
              "span": "test/parse/unary.as:18:5-7",
              "operand": {
                "tag": "Identifier",
                "span": "test/parse/unary.as:18:6",
                "text": "y"
              }
            }
          }
        }
      }
    }
  ]
}
